src/main/java/se/liu/ida/hefquin/engine/data/mappings/impl/EntityMappingImpl.java

```diff
@@ -25,20 +27,67 @@ public EntityMappingImpl( final Graph mappingDescription ) {
 	}
 
 	protected void parseMappingDescription( final Graph mappingDescription ) {
-		// TODO: populate both g2lMap and l2gMap based on the owl:sameAs
+		// populate both g2lMap and l2gMap based on the owl:sameAs
 		// statements in the given RDF graph; assume the the subject of
 		// any owl:sameAs statement is a global URI and the object is the
 		// corresponding local URI
 		final Iterator<Triple> it = mappingDescription.find(Node.ANY, OWL2.sameAs.asNode(), Node.ANY);
 		while ( it.hasNext() ) {
-			// TODO ...
+			final Triple workingTriple = it.next();
+			if ( ! workingTriple.getSubject().isURI() || ! workingTriple.getObject().isURI() ) {
+				continue;
+			}
+			final Set<Node> locals = g2lMap.get(workingTriple.getSubject());
+			if(locals != null) {
+				// There is already at least one local URI for this global URI,
+				// We don't want to override it, so we'll add the new one to the set.
+				locals.add(workingTriple.getObject());
+			} else {
+				// g2lmap doesn't contain any local URI for this global yet
+				final Set<Node> localSet = new HashSet<Node>();
```

Comment:
You need to put this new set into the `g2lMap`.
```suggestion
				final Set<Node> localSet = new HashSet<>();
				g2lMap.put( workingTriple.getSubject(), localSet );
```